include/toolchain/gcc.h

```diff
@@ -359,14 +355,16 @@ do {                                                                    \
 
 #endif /* _ASMLANGUAGE */
 
-#if defined(CONFIG_ARM) && defined(_ASMLANGUAGE)
-#if defined(CONFIG_ISA_THUMB2)
+#if defined(_ASMLANGUAGE)
+#if defined(CONFIG_ARM) && !defined(CONFIG_ARM64)
```

Comment:
@stephanosio could you also explain inline why this is not applicable for AARCH64?

Reply:
As for the obvious, the AArch64 only supports the A64 ISA natively; A32 and T32 are supported only when an ARMv8-A core switches to the AArch32 mode, which exists for legacy application compatibility and should not be applicable to the Zephyr.

As for why there is no AArch64-specific `_ASM_FILE_PROLOGUE`, I think this is something that got overlooked while implementing the AArch64 port.

`toolchain/` headers exist to provide toolchain abstraction; in particular, the ASM macros here allow, at least in theory, the arch ASM source files to be assembler-agnostic. In practice, however, I found that many current implementations, including the AArch64 port, do not make proper use of the toolchain abstraction constructs.

cc @carlocaione

Reply:
`.code` is not supported an AArch64, if that makes sense I can reintroduce `_ASM_FILE_PROLOGUE` to define only `.text` on AArch64.

Reply:
>As for why there is no AArch64-specific _ASM_FILE_PROLOGUE, I think this is something that got overlooked while implementing the AArch64 port.

Yes, that was the source of my confusion actually.

Reply:
>.code is not supported an AArch64, if that makes sense I can reintroduce _ASM_FILE_PROLOGUE to define only .text on AArch64.

@carlocaione could this be considered outside the context of this PR?

Reply:
Yes. And to avoid conflicts I'll post the PR after this is merged.

Reply:
sounds good